pkg/buildscript/mutations.go

```diff
@@ -72,34 +78,44 @@ type RequirementNotFoundError struct {
 	*locale.LocalizedError // for legacy non-user-facing error usages
 }
 
-func (b *BuildScript) removeRequirement(requirement types.Requirement) error {
+// RemoveRequirement will remove any matching requirement. Note that it only operates on the Name and Namespace fields.
+// It will not verify if revision or version match.
+func (b *BuildScript) RemoveRequirement(requirement types.Requirement) error {
```

Comment:
By changing the visibility of this method, it's possible that `UpdateRequirement()` is no longer needed. Please consider removing it.